src/test/java/com/redhat/cloud/policies/app/PoliciesLastTriggeredTest.java

```diff
@@ -0,0 +1,111 @@
+import io.quarkus.test.common.QuarkusTestResource;
+import io.quarkus.test.junit.QuarkusTest;
+import org.hibernate.Session;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import javax.inject.Inject;
+import javax.transaction.Transactional;
+import java.time.Duration;
+import java.time.Instant;
+import java.util.UUID;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotEquals;
+
+@QuarkusTest
+@QuarkusTestResource(TestLifecycleManager.class)
+class PoliciesLastTriggeredTest {
+
+    @Inject
+    Session session;
+
+    private final String testAccountId = "lt-test-account-id";
+    private final String otherTestAccountId = "lt-test-2-account-id";
+
+    @BeforeEach
+    @Transactional
+    void beforeEach() {
+        session.createQuery("DELETE FROM PoliciesHistoryEntry")
+                .executeUpdate();
+    }
+
+    @Test
+    @TestTransaction
+    void testPolicyLastTriggeredIs0ByDefault() {
+        Policy policy = createPolicy(testAccountId, UUID.randomUUID());
+        assertEquals(0, policy.getLastTriggered());
+    }
+
+    @Test
+    @TestTransaction
+    void testPolicyLastTriggeredShouldUpdateToGreatestHistory() {
+        Policy policy = createPolicy(testAccountId, UUID.randomUUID());
+        long lastTime = Instant.now().toEpochMilli();
+
+        createPoliciesHistoryEntry(Instant.now().minus(Duration.ofHours(1L)).toEpochMilli(), policy.id, policy.customerid);
+        createPoliciesHistoryEntry(lastTime, policy.id, policy.customerid);
+        createPoliciesHistoryEntry(Instant.now().minus(Duration.ofDays(28L)).toEpochMilli(), policy.id, policy.customerid);
+
+        policy = getPolicy(policy.id);
+
+        assertEquals(lastTime, policy.getLastTriggered());
+    }
+
+    @Test
+    @TestTransaction
+    void testPolicyLastTriggeredAreSetById() {
+        Policy policy = createPolicy(testAccountId, UUID.randomUUID());
+        Policy policy2 = createPolicy(testAccountId, UUID.randomUUID());
+        Policy policy3 = createPolicy(otherTestAccountId, UUID.randomUUID());
+
+        createPoliciesHistoryEntry(Instant.now().minus(Duration.ofHours(1L)).toEpochMilli(), policy.id, policy.customerid);
+        createPoliciesHistoryEntry(Instant.now().toEpochMilli(), policy.id, policy.customerid);
+        createPoliciesHistoryEntry(Instant.now().minus(Duration.ofDays(28L)).toEpochMilli(), policy.id, policy.customerid);
+
+        policy = getPolicy(policy.id);
+        policy2 = getPolicy(policy2.id);
+        policy3 = getPolicy(policy3.id);
+
+        assertNotEquals(0, policy.getLastTriggered());
+        assertEquals(0, policy2.getLastTriggered());
+        assertEquals(0, policy3.getLastTriggered());
+    }
+
+    private Policy getPolicy(UUID policyId) {
+        // The store procedure does not play well with hibernate.
+        // This should not be a problem with real data as they are updated in separated transactions
```

Comment:
`@TestTransaction` isn't very convenient when you need to create DB data and then use it in the same test. Isn't it possible to get rid of that annotation and simply delete all policies from the `beforeEach` method? `createPolicy` and `createPoliciesHistoryEntry` would then be annotated with `@Transactional` and have an immediate effect on the DB data when invoked.

Reply:
That would need a revamp of these tests. The policies are created by  a migration file. I guess we should move to create it directly as part of the test instead of the migration.

Perhaps a task better suited for a separate PR.

Reply:
For this test is not really needed to delete the policies i guess - so i can play with the annotations

Reply:
If you're busy with something else, this is not so important so we can merge the PR as is 😃

Reply:
Updated - but there was an issue with the `@Transaction` annotation. It wasn't acknowledged as I was getting an exception all the time:
```
javax.persistence.TransactionRequiredException: Transaction is not active, consider adding @Transactional to your method to automatically activate one.

	at io.quarkus.hibernate.orm.runtime.session.TransactionScopedSession.persist(TransactionScopedSession.java:138)
	at io.quarkus.hibernate.orm.runtime.session.ForwardingSession.persist(ForwardingSession.java:53)
	at org.hibernate.Session_5b93bee577ae2f8d76647de04cfab36afbf52958_Synthetic_ClientProxy.persist(Unknown Source)
	at com.redhat.cloud.policies.app.PoliciesLastTriggeredTest.createPolicy(PoliciesLastTriggeredTest.java:99)
	at com.redhat.cloud.policies.app.PoliciesLastTriggeredTest.testPolicyLastTriggeredIs0ByDefault(PoliciesLastTriggeredTest.java:46)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at io.quarkus.test.junit.QuarkusTestExtension.runExtensionMethod(QuarkusTestExtension.java:1008)
	at io.quarkus.test.junit.QuarkusTestExtension.interceptTestMethod(QuarkusTestExtension.java:827)
	at org.junit.jupiter.engine.execution.ExecutableInvoker$ReflectiveInterceptorCall.lambda$ofVoidMethod$0(ExecutableInvoker.java:115)
	at org.junit.jupiter.engine.execution.ExecutableInvoker.lambda$invoke$0(ExecutableInvoker.java:105)
	at ....
```

Reply:
`@Transactional` doesn't work on `private` methods.

Reply:
> `@Transactional` doesn't work on `private` methods.

aha! Thanks!